src/node-jspm-jasmine.js

```diff
@@ -134,12 +142,14 @@ export function runTests(opts, errCallback = function() {}) {
 					const fileKey = getFileKey(getOSFilePath(load.address), process.cwd());
 					// exclude the dependency modules (i.e. libraries) from instrumentation
 					if (coverageFiles[fileKey]) {
+						mkdirp.sync(tempDirectory + fileKey.substring(0, fileKey.lastIndexOf('/')));
 						// put file's transpiled counterpart in temp folder
 						let filename;
 						let sourceMap = '';
+
 						// arrange sourcemaps
 						if (load.metadata.sourceMap) {
-							filename = path.join(tempDirectory, fileKey.replace(/\//g, '|'));
+							filename = path.join(tempDirectory, fileKey);
```

Comment:
@m-a-r-c-e-l-i-n-o interested in your feedback here -- I just mkdirp the directories needed instead of flattening out the directory structure and putting all the files into the root.